### DISCUSSION TITLE
Problem with lidar, and doubt with free joint

### DISCUSSION THREAD

**Discussion by WeekendArchaeo** [google-deepmind/mujoco#2256] [Asking for Help]
### Intro

Hi!

I am a associate professor at Universidad de Málaga, I use MuJoCo for my research on RL on mobile robots. My apologies if my questions are dumb, I've been working on them for a few weeks and reading all the documentation I have found, but still I cannot solve them.

### My setup

I use Mujoco with Python on an Ubuntu 22.04 platform.

My robot is a turtlebot3 burger. The XML model has been adapted from the URDF/Xacro models for this robot that are available elsewhere on the web, and then tuned in order to add some elements like floor, obstacles, and actuators and sensors.

 

### My question

I have tried to create a 360º lidar using the _replicate_ tag. However, as you can see in the image, the lidar does not cover 360 degrees. I have checked the documentation and examples you provide, but unfortunatley, I have no idea of what I'm doing wrong.

I would also like to ask why the free joint I have added to the chassis body is not attached to the center of mass of the robot. I have also noticed that if I remove that free joint, the lidar is reduced to a front cone of about 20º and a rear beam of the same degrees. I do not understand why removing a joint should interfere with the lidar.

![Screenshot at 2024-11-28 11-17-50](https://github.com/user-attachments/assets/654262f4-57b2-451a-949a-74883e449fc7)

### Minimal model and/or code that explain my question

 [turtlebot3burgermjmodel_diff.txt](https://github.com/user-attachments/files/17946852/turtlebot3burgermjmodel_diff.txt)

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
The missing rays are the ones that do not hit anything. If a ray does not hit anything the distance returned is -1 and the ray is not visualized.

**Comment by yuvaltassa** [comment]
If you want your freejoint at the CoM, try using [align](https://mujoco.readthedocs.io/en/latest/XMLreference.html#body-freejoint-align)

**Comment by WeekendArchaeo** [comment]
Thanks a lot for your quick and useful reply :)!!
